### ISSUE TITLE
Licensing expectations/requirements

### ISSUE THREAD

**Issue by rreusser** [scijs/contributing#16]
Are there preferences/requirements for the license of software included in scijs?

It looks like @Planeshifter has transpiled some of GSL into js modules. The attached GNU license could either be seen as an upside or a downside, depending on your views. I won't voice opinions, but the requirements/expectations should at least perhaps be clear.

In other words, I was debating whether to submit my own ODE code to scijs or whether it'd be more effective and future-proof to see if GSL's solvers could be ported. But GSL may or may not be permissible, depending on the goals of this project.

**Comment by Planeshifter** [comment]
To chime in, it seems that the viral nature of the GPL license is quite problematic. In science, not so much, but any business environment will have huge problems with it, so maybe a more permissive license, such as MIT, would be preferable. I just toyed around with GSL to figure out the workflow of porting libraries via emscripten and since the GSL offers a lot of functionality and is quite reputable, it seemed reasonable at the time.

**Comment by rreusser** [comment]
Ah, thanks for the feedback! Yeah, I can see it going either way. Unfortunately, it seems important to at least establish some sort of standard. I think OpenCV has free and non-free sections, so I guess it's not out of the question to take a middle road and let individual components carry their own license.

**Comment by jakirkham** [comment]
If there is a mixture of licenses, it would be nice if somewhere in the docs it was made clear as to what is permissive and what is not. Like an overview of all packages.

**Comment by tab58** [comment]
Personally, I'd like to see all scijs code have an MIT license attached to it. There's no downside for those who want to make some money on something produced with this code, and GPL project creators could use it as well (see [this StackExchange article](http://programmers.stackexchange.com/questions/177768/how-do-i-correctly-sub-license-a-library-that-is-under-the-mit-license)). I hate restrictive licensing for numerical code, especially after reading the draconian Numerical Recipes license (apparently it used to be worse than it is now). GPL for me is another type of licensing that can potentially restrict someone else trying to make a buck on their idea.

Hopefully we can get some sort of consensus on a common license.

**Comment by rreusser** [comment]
Yeah, my approach reading Numerical Recipes is to feel like I need to get a notarized document stating that I promise I really seriously didn't look at the code… That stuff drives me crazy. MIT across the board sounds great to me—except in the event that we pull in code that carries a separate license, in which case we need to make it very clear that a separate license may apply, although such cases will hopefully be rare and can be dealt with on a case by case basis.

I don't _not_ support WTFPL, but the additional terms of the MIT license are perfectly fine with me. Unless I'm missing subtlety, the terms are basically just: 1. Do whatever you want. 2. Give credit, and 3. No warranty. I have no issues with any of that.

So MIT across the board unless a specific case requires otherwise?
